packages/opencensus-core/test/test-derived-gauge.ts

```diff
@@ -73,15 +68,19 @@ describe('DerivedGauge', () => {
       map.set('key', 'value');
       instance.createTimeSeries(LABEL_VALUES_200, map);
       map.set('key1', 'value1');
+
+      now = Date.now();
       let metric = instance.getMetric();
       assert.notEqual(metric, null);
       assert.deepStrictEqual(metric.descriptor, expectedMetricDescriptor);
       assert.equal(metric.timeseries.length, 1);
-      assert.deepStrictEqual(
-          metric.timeseries, [{
-            labelValues: LABEL_VALUES_200,
-            points: [{value: 2, timestamp: {nanos: 1e7, seconds: 1000}}]
-          }]);
+
+      const [timeseries] = metric.timeseries;
+      const [points] = timeseries.points;
+      assert.deepStrictEqual(timeseries.labelValues, LABEL_VALUES_200);
+      assert.equal(points.value, 2);
+      assert.equal(Math.floor(now / 1e3), points.timestamp.seconds);
```

Comment:
What would you think about mocking `Date.now` and `process.hrtime` for the test (setting them and unsetting them before/after)? That way if someone sets a breakpoint in the test and some time elapses it will still pass and you can also assert exact values for nanoseconds.

Reply:
What if you had something like this:

```
let hrtime = process.hrtime;
let hrtimeOrigin: [number, number] = [0, 0];
let hrtimeRefSeconds = 0;
let hrtimeRefNanos = 0;

/** This is used to enable tests to mock process.hrtime while still allow us to cache it. */
function resetHrtimeFunctionCache() {
   hrtime = process.hrtime;
}

function setHrtimeReference() {
  resetHrtimeFunctionCache();
  hrtimeOrigin = hrtime();
  const refTime = Date.now();
  hrtimeRefSeconds = ...
  hrtimeRefNanos = ...
}

setHrtimeReference();

export TEST_ONLY = {setHrtimeReference};

.....

// In the test

import {TEST_ONLY, ...} from '../common/time-utils.ts';

...

let realHrtimeFn = process.hrtime;
let realNowFn = Date.now;
beforeEach(() => {
   process.hrtime = () => [55, 22];
   Date.now = () => 14500000;
    // Force the clock to recalibrate the time offset with the mocked time
   TEST_ONLY.setHrtimeReference();
   // Simulate that now some time passed ...   
   process.hrtime = () => [66, 33];
   TEST_ONLY.resetHrtimeFunctionCache();
});

afterEach(() => {
   process.hrtime = realHrtimeFn;
   Date.now = realDateNow;
   // Reset the hrtime reference so that it uses a real clock again.
   TEST_ONLY.setHrtimeReference();
});
```

If this feels too involved, I'm OK with skipping it but wanted to put it there as an option. If you go this route, it could be helpful to export these as some kind of time testing helpers to easily set/unset the mocked time.

Reply:
@draffensperger Awesome, thanks for the help. I have modified the tests, PTAL.